### PR TITLE
Switch from cmdline GlobalInstance to app registry

### PR DESCRIPTION
**Pull request by ghjm**
Use of a naïve global variable for registering cmdline configurations leads to issue https://github.com/ansible/receptor/issues/334.  To fix this, I have added a global app registry to cmdline, which allows registering groups of configuration items under a namespace.  The main application can then decide which items from this namespace it wants to add to its actual command line.

I have also changed many of the cmdline config objects from public to private.  Their members need to be public so cmdline can see them, but the objects themselves can be private.  Making them private, among other things, prevents them from showing up in godoc.

### CODE REVIEW

**Comment by ghjm** [comment]
I don't have permissions to re-start the CI checks.  They succeed locally:

```
(graham-dev) ✔ ~/git/receptor [cmdline_registry|✔]
20:55 $ SKIP_KUBE=1 make ci
go fmt...................................................................Passed
go lint..................................................................Passed
Running Go builds...
?       github.com/project-receptor/receptor/cmd        [no test files]
?       github.com/project-receptor/receptor/example    [no test files]
ok      github.com/project-receptor/receptor/pkg/backends       0.275s
?       github.com/project-receptor/receptor/pkg/certificates   [no test files]
?       github.com/project-receptor/receptor/pkg/controlsvc     [no test files]
?       github.com/project-receptor/receptor/pkg/framer [no test files]
?       github.com/project-receptor/receptor/pkg/logger [no test files]
ok      github.com/project-receptor/receptor/pkg/netceptor      3.511s
?       github.com/project-receptor/receptor/pkg/randstr        [no test files]
?       github.com/project-receptor/receptor/pkg/services       [no test files]
?       github.com/project-receptor/receptor/pkg/tickrunner     [no test files]
ok      github.com/project-receptor/receptor/pkg/utils  1.003s
?       github.com/project-receptor/receptor/pkg/version        [no test files]
ok      github.com/project-receptor/receptor/pkg/workceptor     5.618s
ok      github.com/project-receptor/receptor/tests/functional/cli       0.500s
?       github.com/project-receptor/receptor/tests/functional/lib/mesh  [no test files]
?       github.com/project-receptor/receptor/tests/functional/lib/receptorcontrol       [no test files]
ok      github.com/project-receptor/receptor/tests/functional/lib/utils 0.005s
ok      github.com/project-receptor/receptor/tests/functional/mesh      82.995s
cd receptorctl && ../receptorctl-test-venv/bin/pytest tests/tests.py
============================================= test session starts =============================================
platform linux -- Python 3.9.5, pytest-6.1.1, py-1.10.0, pluggy-0.13.1
rootdir: /home/graham/git/receptor/receptorctl
collected 5 items

tests/tests.py .....                                                                                    [100%]

============================================== 5 passed in 0.78s ==============================================
All done
```

File: pkg/backends/tcp.go [hunk 1]
```diff
@@ -251,24 +251,24 @@ func (cfg TCPListenerCfg) Run() error {
 	return nil
 }
 
-// TCPDialerCfg is the cmdline configuration object for a TCP dialer
-type TCPDialerCfg struct {
+// tCPDialerCfg is the cmdline configuration object for a TCP dialer
+type tCPDialerCfg struct {
```
Comment (shanemcd) [inline pkg/backends/tcp.go:255]: Was this change intentional?

File: pkg/backends/tcp.go [hunk 2]
```diff
@@ -210,8 +210,8 @@ func (ns *TCPSession) Close() error {
 // Command line
 // **************************************************************************
 
-// TCPListenerCfg is the cmdline configuration object for a TCP listener
-type TCPListenerCfg struct {
+// tCPListenerCfg is the cmdline configuration object for a TCP listener
+type tCPListenerCfg struct {
```
Comment (shanemcd) [inline pkg/backends/tcp.go]: Is this really the canonical way to name these variables in Go? It looks so strange...

Comment (shanemcd) [reply pkg/backends/tcp.go]: What about `tcpListenerCfg`?

Comment (ghjm) [reply pkg/backends/tcp.go]: You mean with the small first letter?  Yes, I agree it looks deeply weird.  But that's how you do it in Go.  A small first letter means private, capital first letter means public.

Comment (ghjm) [reply pkg/backends/tcp.go]: The linter is usually very opinionated about these things, but doesn't complain about `tcpListenerCfg`.  I don't care, I just don't want linter errors.  😁

I'll change them to that style.

**Review by fosterseth — APPROVED** [review]
